packages/components/src/composite/legacy.tsx

```diff
@@ -0,0 +1,301 @@
+	baseId?: string;
+	unstable_virtual?: boolean;
+	rtl?: boolean;
+	orientation?: Orientation;
+	currentId?: string;
+	loop?: boolean | Orientation;
+	wrap?: boolean | Orientation;
+	shift?: boolean;
+}
+
+export interface CompositeState {
+	store: CompositeStore;
+
+	up: () => void;
+	down: () => void;
+	first: () => void;
+	last: () => void;
+	previous: () => void;
+	next: () => void;
+	move: ( id: string | undefined ) => void;
+
+	baseId: IdState[ 'baseId' ];
+	setBaseId: ( v: CompositeState[ 'baseId' ] ) => void;
+	currentId: string | undefined;
+	setCurrentId: ( v: CompositeState[ 'currentId' ] ) => void;
+	orientation: Orientation | undefined;
+	setOrientation: ( v: CompositeState[ 'orientation' ] ) => void;
+	rtl: boolean;
+	setRTL: ( v: CompositeState[ 'rtl' ] ) => void;
+	loop: boolean | Orientation;
+	setLoop: ( v: CompositeState[ 'loop' ] ) => void;
+	shift: boolean;
+	setShift: ( v: CompositeState[ 'shift' ] ) => void;
+	wrap: boolean | Orientation;
+	setWrap: ( v: CompositeState[ 'wrap' ] ) => void;
+}
+
+type FilteredCompositeStateProps = Pick< CompositeState, 'store' | 'baseId' >;
+
+type PartialCompositeState = Partial<
+	Omit< CompositeState, keyof FilteredCompositeStateProps >
+> &
+	FilteredCompositeStateProps;
+
+export type CompositeStateProps =
+	| { state: PartialCompositeState }
+	| ( PartialCompositeState & { state?: never } );
+
+interface IdState {
+	baseId: string;
+	setBaseId: React.Dispatch< React.SetStateAction< string > >;
+}
+
+function showDeprecationMessage( ...props: Parameters< typeof deprecated > ) {
+	if ( 'test' !== process.env.NODE_ENV ) {
+		deprecated( ...props );
+	}
+}
+
+const idMap = new Map< string, React.MutableRefObject< number > >();
+
+function useBaseId(
+	initialState: Pick< Partial< IdState >, 'baseId' > = {}
+): IdState {
+	const { baseId: initialBaseId } = initialState;
+	const [ baseId, setBaseId ] = useState(
+		() =>
+			initialBaseId ||
+			// eslint-disable-next-line no-restricted-syntax
+			`id-${ Math.random().toString( 32 ).substring( 2, 8 ) }`
+	);
+	idMap.set( baseId, useRef( 0 ) );
+	return { baseId, setBaseId };
+}
+
+function useId( baseId: string, preferredId?: string ) {
+	const counter = idMap.get( baseId ) ?? { current: 0 };
+
+	const [ id, setId ] = useState( () => {
+		if ( preferredId ) return preferredId;
+		return `${ baseId }-${ ++counter.current }`;
+	} );
+
+	return { id, setId };
+}
+
+function useMapInitialStateToStoreProps(
+	initialState: InitialState = {}
+): CompositeStoreProps & IdState {
+	const {
+		currentId: defaultActiveId,
+		orientation,
+		rtl = false,
+		loop: focusLoop = false,
+		wrap: focusWrap = false,
+		shift: focusShift = false,
+		// eslint-disable-next-line camelcase
+		unstable_virtual: virtualFocus,
+		...idState
+	} = initialState;
+
+	return {
+		...useBaseId( idState ),
+		defaultActiveId,
+		rtl,
+		orientation,
+		focusLoop,
+		focusShift,
+		focusWrap,
+		virtualFocus,
+
+		// TODO?
+		includesBaseElement: false,
```

Comment:
I think this can be omitted. Ariakit will infer this value from `defaultActiveId`/`activeId` the same way Reakit does.